node-upstream-tests/run_tests.js

```diff
@@ -0,0 +1,387 @@
+'use strict'
```

Comment:
This file could also benefit from being split up in a few files. There are a lot of functions, variable declarations and classes declared seemingly at random places across the file which makes it difficult to read. Otherwise most of the logic looks good to me.